--- ppl-0.11.2/src/mp_std_bits.cc.bero	2013-02-13 00:45:10.866844896 +0100
+++ ppl-0.11.2/src/mp_std_bits.cc	2013-02-13 00:45:31.066602270 +0100
@@ -46,7 +46,9 @@ const bool std::numeric_limits<mpz_class
 const bool std::numeric_limits<mpz_class>::is_bounded;
 const bool std::numeric_limits<mpz_class>::is_modulo;
 const bool std::numeric_limits<mpz_class>::traps;
+#if __GNU_MP_RELEASE < 50100
 const bool std::numeric_limits<mpz_class>::tininess_before;
+#endif
 const std::float_round_style std::numeric_limits<mpz_class>::round_style;
 
 const bool std::numeric_limits<mpq_class>::is_specialized;
@@ -69,5 +71,7 @@ const bool std::numeric_limits<mpq_class
 const bool std::numeric_limits<mpq_class>::is_bounded;
 const bool std::numeric_limits<mpq_class>::is_modulo;
 const bool std::numeric_limits<mpq_class>::traps;
+#if __GNU_MP_RELEASE < 50100
 const bool std::numeric_limits<mpq_class>::tininess_before;
+#endif
 const std::float_round_style std::numeric_limits<mpq_class>::round_style;
--- ppl-0.11.2/src/mp_std_bits.defs.hh.bero	2013-02-13 00:37:49.171976954 +0100
+++ ppl-0.11.2/src/mp_std_bits.defs.hh	2013-02-13 00:43:00.570394496 +0100
@@ -39,6 +39,7 @@ void swap(mpz_class& x, mpz_class& y);
 #endif // defined(PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS)
 void swap(mpq_class& x, mpq_class& y);
 
+#if __GNU_MP_RELEASE < 50100 // Already defined in gmp++ 5.1
 namespace std {
 
 #ifdef PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS
@@ -164,6 +165,7 @@ public:
 };
 
 } // namespace std
+#endif
 
 #include "mp_std_bits.inlines.hh"
 
